src/test/scala/org/broadinstitute/dsde/firecloud/service/NihApiServiceSpec.scala

```diff
@@ -35,63 +48,101 @@ class NihApiServiceSpec extends ApiServiceSpec {
-    Post("/nih/callback", validJwtOnWhitelist) ~> dummyUserIdHeaders(services.thurloeDao.linkedButExpiredUserId) ~> sealRoute(services.nihRoutes) ~> check {
+    Post("/nih/callback", firecloudDevJwt) ~> dummyUserIdHeaders(toLink) ~> sealRoute(services.nihRoutes) ~> check {
       status should equal(OK)
     }
 
     //verify that their link expiration has been updated
-    Get("/nih/status") ~> dummyUserIdHeaders(services.thurloeDao.linkedButExpiredUserId) ~> sealRoute(services.nihRoutes) ~> check {
+    Get("/nih/status") ~> dummyUserIdHeaders(toLink) ~> sealRoute(services.nihRoutes) ~> check {
       status should equal(OK)
       val linkExpireTime = responseAs[NihStatus].linkExpireTime.get
 
       assert(linkExpireTime >= DateUtils.nowMinus1Hour) //link expire time is fresh
       assert(linkExpireTime <= DateUtils.nowPlus30Days) //link expire time is approx 30 days in the future
     }
-    assert(services.rawlsDao.groups(dbGapAuthorizedUsersGroupName).contains(services.thurloeDao.linkedButExpiredUserId))
+    assert(services.rawlsDao.groups(tcgaDbGaPAuthorized).contains(toLink))
+    assert(services.rawlsDao.groups(targetDbGaPAuthorized).contains(toLink))
   }
 
   /* Test scenario:
-     1 user that is linked but their TCGA access has expired. they should be removed
-     1 user that is linked but they have no expiration date stored in their profile. they should be removed
-     1 user that is linked and has active TCGA access. they should remain in the dbGapAuthorizedUsers group
+     1 user that is linked but their TCGA access has expired. they should be removed from the TCGA group
+     1 user that is linked but their TARGET access has expired. they should be removed from the TARGET group
+     1 user that is linked but they have no expiration date stored in their profile. they should be removed from the TCGA group
+     1 user that is linked but their TCGA and TARGET access has expired. they should be removed from the TARGET and TCGA groups
+     1 user that is linked and has active TCGA access. they should remain in the TCGA group
+     1 user that is linked and has active TARGET access. they should remain in the TARGET group
+     1 user that is linked and has active TARGET & TCGA access. they should remain in the TARGET and TCGA groups
    */
-  it should "return NoContent and properly sync the whitelist for users of different link statuses" in withDefaultApiServices { services =>
+  it should "return NoContent and properly sync the whitelist for users of different link statuses across whitelists" in withDefaultApiServices { services =>
     Post("/sync_whitelist") ~> sealRoute(services.syncRoute) ~> check {
       status should equal(NoContent)
     }
-    assert(services.rawlsDao.groups(dbGapAuthorizedUsersGroupName).equals(Set(services.thurloeDao.linkedUserId)))
+    services.rawlsDao.groups(tcgaDbGaPAuthorized) should contain theSameElementsAs Set(services.thurloeDao.TCGA_AND_TARGET_LINKED, services.thurloeDao.TCGA_LINKED)
```

Comment:
`assertSameElements` please, even though it doesn't make a difference in this case